app/components/visibility-setting-list.js

```diff
@@ -0,0 +1,109 @@
+import Component from '@ember/component';
+import { computed } from '@ember/object';
+import { reads, empty, not } from '@ember/object/computed';
+import KeyboardShortcuts from 'ember-keyboard-shortcuts/mixins/component';
+
+export default Component.extend(KeyboardShortcuts, {
+  classNames: ['visibility-setting-list'],
+
+  keyboardShortcuts: {
+    'esc': 'closeConfirmationModal'
+  },
+
+  // An example item that might be in the options object:
+  // private: {
+  //   displayValue: 'you',
+  //   description: 'Do not allow everyone to see insights from your private builds',
+  //   modalText: 'Do not allow everyone to see my private insights',
+  // }
+  //
+  // The key is used to match selected and currentSelection up with the correct details
+  // `displayValue` is used to generate text for the modal
+  // `description` is for the label next to the radio button
+  // `modalText` can be used to override the generated modal text
+  options: computed(() => ({})),
+  optionKeys: computed('options', function () { return Object.keys(this.options); }),
+  optionValues: computed('options', function () { return Object.values(this.options); }),
+  isShowingConfirmationModal: false,
+  isEmpty: empty('optionKeys'),
+  isVisible: not('isEmpty'),
+
+  selected: '',
+  currentSelection: reads('selected'),
+  currentSelectionIndex: computed('currentSelection', 'optionKeys.[]', function () {
+    return this.optionKeys.findIndex((slug) => slug === this.currentSelection);
+  }),
+  selectedIndex: computed('selected', 'optionKeys.[]', function () {
+    return this.optionKeys.findIndex((slug) => slug === this.selected);
+  }),
+  change: computed('selectedIndex', 'currentSelectionIndex', function () {
+    return this.currentSelectionIndex - this.selectedIndex;
+  }),
+
+  selectionDetails: computed('currentSelection', 'options', function () {
+    const { options, currentSelection } = this;
+
+    if (!options.hasOwnProperty(currentSelection)) {
+      return {};
+    }
+
+    return options[currentSelection];
+  }),
+
+  modalHeaderText: computed('change', function () {
+    let operation;
+    if (this.change < 0) {
+      operation = 'Restrict';
+    } else if (this.change > 0) {
+      operation = 'Increase';
+    } else {
+      operation = 'Update';
+    }
+
+    return `${operation} visibility of your private build insights`;
+  }),
```

Comment:
I would suggest to move this to template. You could define couple computed properties and then use them in template:
```js
isChangeNegative: lt('change', 0),
isChangePositive: gt('change', 0)
```
```hbs
{{#if isChangeNegative}}
  Restrict
{{else if isChangePositive}}
  Increase
{{else}}
  Update
{{/if}}
visibility of your private build insights
```
Copy is usually presentation logic and it should belong to templates whenever possible.

Reply:
That makes a lot of sense and seems way better! Change made :)